### PR TITLE
Transport certs jwt

### PR DESCRIPTION
**Pull request by IgorBalog-Eng**
Added logic for including ssl-server hash in jwt
TransportCertsManager for comparing referringConnector from jwt with SubjectAlternativeName from certificates from trustStore
new property application.extendedTokenValidation=false for enabling/disabling extended jwt validation

### CODE REVIEW

**Review by marest94 — APPROVED** [review]
As far as my knowledge of the TRUE Connector and certificates goes, this looks fine.

File: src/test/resources/application.properties [hunk 1]
```diff
@@ -57,6 +60,7 @@ application.fetchTokenOnStartup=false
 application.daps.orbiter.privateKey=
 application.daps.orbiter.password=
 application.dapsVersion=orbiter
+#application.targetDirectory=./src/test/resources/
```
Comment (davidjovanovic) [inline src/test/resources/application.properties:63]: similar as above one, left on purpose or accidentally?

File: src/test/java/it/eng/idsa/businesslogic/processor/receiver/ReceiverSendDataToDataAppProcessorTest.java [hunk 2]
```diff
@@ -104,7 +104,7 @@ public void processFormSuccess() throws Exception {
 	}
 	
 	@Test
-	public void processHttpHeadermSuccess() throws Exception {
+	public void processHttpHeaderSuccess() throws Exception {
```
Comment (davidjovanovic) [inline src/test/java/it/eng/idsa/businesslogic/processor/receiver/ReceiverSendDataToDataAppProcessorTest.java:107]: nice one

File: src/main/java/it/eng/idsa/businesslogic/service/impl/TransportCertsManager.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+				logger.info("Truststore not configured, cannot calculate TransportCertsSha256");
+			}
+		} finally {
+			if (jksTrustStoreInputStream != null) {
+				jksTrustStoreInputStream.close();
+			}
+		}
+	}
+
+	public String getConnectorTransportCetsSHa() {
```
Comment (davidjovanovic) [inline src/main/java/it/eng/idsa/businesslogic/service/impl/TransportCertsManager.java:63]: CetsSHa typo big H

File: src/main/java/it/eng/idsa/businesslogic/service/impl/DapsV2ServiceImpl.java [hunk 4]
```diff
@@ -172,4 +216,21 @@ String getJwTokenInternal() {
...
 	}
+
+	private String createCertsShaClaim() {
+//		{
+//		    "access_token": {
+//		        "transportCertsSha256": {
+//		            "value": ["ksjdhvs87h3w4fjhsf87hkjvs", "qz47djs87h3w4fjhsf87hg57d"]
+//		        }
+//		    }
+//		}
```
Comment (davidjovanovic) [inline src/main/java/it/eng/idsa/businesslogic/service/impl/DapsV2ServiceImpl.java:227]: Leftover code or left on purpose?

**Pull request merged**